core/common/src/main/java/alluxio/uri/Authority.java

```diff
@@ -23,6 +23,8 @@
  */
 public interface Authority extends Comparable<Authority>, Serializable {
   Logger LOG = LoggerFactory.getLogger(Authority.class);
+  Pattern LOGICAL_MASTER_AUTH = Pattern.compile("^([^:,;@]+)$");
+  Pattern ZOOKEEPER_LOGICAL_MASTER_AUTH = Pattern.compile("^zk@([^:,;@]+)$");
```

Comment:
@ZhuTopher I can't get it. Can you provide a test case just like written in `AuthorityTest`.I don't think there is a problem with handling uri like zk@logical:19998 currently, this uri should't be treated as a logical uri.

Reply:
The issue I'm referring to isn't that it won't match, it's that the port information won't be in a separate [capture group](https://www.regular-expressions.info/brackets.html). For example the groups for `SINGLE_MASTER_AUTH = Pattern.compile("^([^:,;]+):(\\d+)$")` let us easily extract the host and the port from the string:
```
new SingleMasterAuthority(matcher.group(1), Integer.parseInt(matcher.group(2)));
```

For my second point, I was saying that the regex pattern is very generic (it matches a lot of other patterns). What distinguishes a logical Authority URI from the existing `ZOOKEEPER_AUTH` (or logical master Authority URI from `SINGLE_MASTER_AUTH`)? Whatever the difference is, I'd like the regexes to accurately capture that as opposed to relying only on the ordering of matching in `Authority#fromString()`.
- Also, could you clarify a bit what exactly the difference is between a logical Authority and the existing ones is?

Reply:
@ZhuTopher Logical authority don't have port. It's the difference between a logical authority and a real authority.  `ZOOKEEPER_LOGICAL_MASTER_AUTH`  and `LOGICAL_MASTER_AUTH` will match authority like `logical` and `zk@logical`. `ZOOKEEPER_AUTH ` and `SINGLE_MASTER_AUTH` will match authority like `real:19998` and `zk@real:2181`. These two regex patterns will not be confused with each other.

Reply:
Thanks for the explanation! Ah my bad, you're right. Your logical authority regex will not match ports, so that looks good. Let me review the linked issues a bit more and then I'll re-review this PR.